config/credentials.go

```diff
@@ -0,0 +1,265 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/kiali/kiali/log"
+)
+
+// credentialCache caches file-based credentials and watches for changes
+type credentialCache struct {
+	mu          sync.RWMutex
+	cache       map[string]string // filepath -> cached content
+	watchedDirs map[string]struct{}
+	watcher     *fsnotify.Watcher
+	done        chan struct{}
+}
+
+var (
+	watchedCredentialFileCache     *credentialCache
+	watchedCredentialFileCacheOnce sync.Once
+)
```

Comment:
Do these need to be global variables? Typically it makes things harder to test and harder to modify in the future and adds more side effects. Maybe the credential cache can live on the `Config` object? Might still need to be a "ReadCredential" method to avoid things that expect the `Config` object to not be modified.

Reply:
globals removed. all state encapsulated in the CredentialManager struct, and each config.Config instance holds its own *CredentialManager for easy testing

CertPool supports auto‑rotation via fsnotify; callers just call CertPool() (or APIs that use it, like httputil.CreateTransport) and no longer need to manually call any ReadCredential‑style helper.